samples/bluetooth/hci_usb/README.rst

```diff
@@ -21,3 +21,28 @@ This sample can be found under :zephyr_file:`samples/bluetooth/hci_usb` in the
 Zephyr tree.
 
 See :ref:`bluetooth samples section <bluetooth-samples>` for details.
+
+Notes on HCI over USB
+*********************
+
+USB is a serial protocol. However, the implementation of the peripheral can
+expose multiple endpoints, in effect re-ordering serial data into per-endpoint
+receive queues.
+
+The Bluetooth specification defines separate endpoints per HCI packet type.
+
+As the HCI protocol was not originally designed with re-ordering in mind, using
+this transport may expose the application to deadlocks and general unexpected
+behavior.
+
+Because of this, the industry-standard way to expose a Bluetooth Controller over
+USB is now to use the H4 HCI transport over virtual serial (i.e. CDC class).
+
+See the :zephyr:code-sample:`HCI UART sample <bluetooth-hci-uart-sample>`
```

Comment:
This would require the sample to use the new-ish `zephyr:code-sample` directive, otherwise you should link to it as a "regular" `ref`.
I would really welcome the migration of the Bluetooth samples to `code-sample`, so maybe this PR is an opportunity to do it in staged phases? See how e.g. new BAP sample will nicely show up in API reference .. https://builds.zephyrproject.io/zephyr/pr/68228/docs/connectivity/bluetooth/api/bap.html

Reply:
I see. I can add another commit (in that PR) doing that for the hci_xx samples if that's alright.

Reply:
Of course!